rust/projects/file-io/tests/tests.rs

```diff
@@ -116,12 +140,40 @@ fn get_non_existent_value() -> Result<()> {
 
     // Open from disk again and check persistent data
     drop(store);
-    let store = KvStore::open(temp_dir.path())?;
+    let mut store = KvStore::open(temp_dir.path())?;
     assert_eq!(store.get("key2".to_owned())?, None);
 
     Ok(())
 }
 
+// Insert lots of updates and verify the correctness.
+// Please adjust your parameters to make your KvStore trigger
+// compaction more often.
+#[test]
+fn compaction() -> Result<()> {
+    // map in memory for verifying correctness
+    let mut map = HashMap::new();
+    let mut rng = thread_rng();
+
+    let temp_dir = TempDir::new().expect("unable to create temporary working directory");
+    let mut store = KvStore::open(temp_dir.path())?;
+
+    for _ in 0..10_000 {
+        let key = format!("key{}", rng.gen::<u8>());
+        let value = format!("{}", rng.gen::<u64>());
+        map.insert(key.clone(), value.clone());
+        store.set(key, value)?;
+
+        if rng.gen::<f64>() < 0.1 {
+            for (k, v) in &map {
+                assert_eq!(store.get(k.clone())?.as_ref(), Some(v));
+            }
+        }
+    }
+
+    Ok(())
+}
```

Comment:
This method doesn't look to be testing compaction.

Reply:
I agree. But I just don't know how to test it.

I'm thinking about making `compact` a public function first, and then check the correctness of values in store. 

However, only correctness is tested. Implementors can just leave `compact` blank. Inserting a large amount of data is not practical. It will take too much time.

Reply:
OK, we'll think about it more later.